security/advancedtls/advancedtls.go

```diff
@@ -0,0 +1,377 @@
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package advancedtls is a utility library containing functions to construct
+// credentials.TransportCredentials that can perform credential reloading and custom
+// server authorization.
+// All APIs in this package are EXPERIMENTAL.
+package advancedtls
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net"
+	"syscall"
+	"time"
+
+	"google.golang.org/grpc/credentials"
+)
+
+type sysConn = syscall.Conn
```

Comment:
Worth a comment here why this is needed.  Also, I'd move this down to where it's needed.